caddyfile_test.go

```diff
@@ -236,6 +237,26 @@ func Test_parseSupervisor(t *testing.T) {
 				}
 			`,
 		},
+		{
+			name: "replicas zero",
+			givenCaddyfile: `
+				supervisor {
+				  php-fpm --fpm-config=fpm-8.0.2.ini {
+					replicas 0
+				  }
+				}
+			`,
+			expectJson: `
+				{
+					"supervise": [
+						{
```

Comment:
What about just removing the command in the resulting the json? What do you think @francislavoie?

Reply:
I think there's value in allowing JSON to set it to 0 if you use the config API primarily, cause then you can make a single API call to turn off the service by setting replicas to 0.

Actually, is that handled? Is this app using global state for tracking the services it's running? I think it should, cause that would allow reloading config to not tear down and start services every time which can be expensive and not necessarily needed. 🤔

Reply:
The question was more about skipping the entry when converting from Caddyfile to json.

For now the app is not in sync with the global state, everything is done at startup. Do you have some resources I can check to implement this? How can I watch the current config?

Reply:
I'm just on my phone so I'm not looking at the whole codebase right now, but the idea is that you keep global references to the replicas you are running, with the command/settings used to launch it, and store a atomic global counter to know if Caddy is reloading or stopping.

The counter: +1 on Start(), -1 on Stop(), cause on reload there's two instances of your config running at the same time for a short period of time. So if on Stop() the counter is at 2 then this is just a reload cause there's another config that was started so Stop() can skip teardown. If during Stop() the counter is at 1 then Caddy is shutting down.

Then if it's a reload, you'd count how many running instances you have which do and don't share the same config as the previous config (compare new config to the global running instances, would need a Compare() defined for it). If then the replicas count of the new config is higher, then you spin up new instances to match. If it's lower then turn some off. If the config is different, turn those off (and turn on new ones to replace it if replicas count asks for it).

Reply:
So much respect for writing this on a phone :D 

Thanks for the tips!